ios/wrappers/logging/CioLoggingEmitter.swift

```diff
@@ -0,0 +1,96 @@
+
+import React
+import CioInternalCommon
+
+typealias CioLogger = CioInternalCommon.Logger
+
+@objc(CioLoggingEmitter)
+class CioLoggingEmitter: RCTEventEmitter {
+    
+    fileprivate static var eventName = "CioLogEvent"
+    
+    fileprivate var hasObservers = false
+    
+    // Requires adding requiresMainQueueSetup method
+    // since it overrides init
+    @objc static override func requiresMainQueueSetup() -> Bool {
+        return true // Return true if the module must be initialized on the main queue
+    }
+    override func startObserving() {
+        hasObservers = true
+    }
+    
+    override func stopObserving() {
+        hasObservers = false
+    }
+    
+    override func supportedEvents() -> [String] {
+        [Self.eventName]
+    }
+    
+    override var methodQueue: dispatch_queue_t! {
+        DispatchQueue(label: Self.moduleName())
+    }
+}
+
+
+class CioLoggerWrapper: CioLogger {
+    private(set) var logLevel: CioLogLevel
+    private var moduleRegistry: RCTModuleRegistry
```

Comment:
not sure what the lifecycle is like for this module, should it be `weak`?

Reply:
This is initialised when the SDK is initialised [here](https://github.com/customerio/customerio-reactnative/blob/be3665994ae755572b2db01b9507ed6f3ddab08b/ios/wrappers/CioRctWrapper.swift#L16). But it would make sense to use `weak` in this case.